src/HLSLTree.cpp

```diff
@@ -1409,139 +1409,5 @@ void HideUnusedArguments(HLSLFunction * function)
     }
 }
 
-
-class FindFunctionCallVisitor : public HLSLTreeVisitor
```

Comment:
Can you explain why this approach isn't valid? I'm not advocating it, just want to understand. Ideally I think the necessary functions should be listed during traversal and prepended to the generated code, instead of traversing the tree multiple times for each function.